service/awsconfig/v5/resource/cloudformation/update.go

```diff
@@ -64,24 +84,104 @@ func (r *Resource) newUpdateChange(ctx context.Context, obj, currentState, desir
 		return cloudformation.CreateStackInput{}, microerror.Mask(err)
 	}
 
-	r.logger.LogCtx(ctx, "debug", "finding out if the main stack should be updated")
-
-	updateState := cloudformation.UpdateStackInput{}
-
-	if currentStackState.Name != "" && !reflect.DeepEqual(desiredStackState, currentStackState) {
-		r.logger.LogCtx(ctx, "debug", "main stack should be updated")
+	if shouldScale(currentStackState, desiredStackState) {
+		r.logger.LogCtx(ctx, "debug", "main stack has to be scaled")
 
-		mainTemplate, err := r.getMainGuestTemplateBody(customObject)
+		updateState, err := r.computeUpdateState(customObject, desiredStackState)
 		if err != nil {
-			return nil, microerror.Mask(err)
+			return cloudformation.CreateStackInput{}, microerror.Mask(err)
 		}
-		updateState.StackName = aws.String(desiredStackState.Name)
-		updateState.TemplateBody = aws.String(mainTemplate)
-		// CAPABILITY_NAMED_IAM is required for updating IAM roles (worker policy).
-		updateState.Capabilities = []*string{
-			aws.String(namedIAMCapability),
+
+		return updateState, nil
+	} else {
+		r.logger.LogCtx(ctx, "debug", "main stack has not to be scaled")
+	}
+
+	// We enable/disable updates in order to enable them our test installations
+	// but disable them in production installations. That is useful until we have
+	// full confidence in updating guest clusters.
+	if updateallowedcontext.IsUpdateAllowed(ctx) {
+		r.logger.LogCtx(ctx, "debug", "finding out if the main stack has to be updated")
+
+		if shouldUpdate(currentStackState, desiredStackState) {
```

Comment:
We only update in case the guest clusters should be updated. This is independent from scaling.

Reply:
Nice!